internal/pkg/service/appsproxy/dataapps/api/config.go

```diff
@@ -80,6 +112,11 @@ func (a *API) GetAppConfig(appID AppID, eTag string) request.APIRequest[*AppConf
 			// Add MaxAge
 			result.maxAge = maxAge
 
+			// Sort rules, longest value/path first
+			slices.SortStableFunc(result.AuthRules, func(a, b Rule) int {
+				return strings.Compare(b.Value, a.Value)
+			})
```

Comment:
@jachym-tousek-keboola it match your original implementation :slightly_smiling_face: 

But good point, we should address it.

> Patterns name fixed, rooted paths, like "/favicon.ico", or rooted subtrees, like "/images/" (note the trailing slash). **Longer patterns take precedence over shorter ones**, so that if there are handlers registered for both "/images/" and "/images/thumbnails/", the latter handler will be called for paths beginning "/images/thumbnails/" and the former will receive requests for any other paths in the "/images/" subtree.

https://stackoverflow.com/a/55216191

Reply:
It is part of a changes in Go 1.22 (one more reason to not use it in our case), but `Precedence` seems to be same.
The sentences from stackoverflow are no more there.

```
The pattern syntax and matching behavior of ServeMux changed significantly in Go 1.22. 
```

```
Precedence
If two or more patterns match a request, then the most specific pattern takes precedence. A pattern P1 is more specific than P2 if P1 matches a strict subset of P2’s requests; that is, if P2 matches all the requests of P1 and more. If neither is more specific, then the patterns conflict. There is one exception to this rule, for backwards compatibility: if two patterns would otherwise conflict and one has a host while the other does not, then the pattern with the host takes precedence. If a pattern passed [ServeMux.Handle](https://pkg.go.dev/net/http#ServeMux.Handle) or [ServeMux.HandleFunc](https://pkg.go.dev/net/http#ServeMux.HandleFunc) conflicts with another pattern that is already registered, those functions panic.
```

https://pkg.go.dev/net/http#hdr-Precedence